src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -200,9 +202,53 @@ private void AddCompilationRoots()
                     AddCompilationRootsForMainMethod(module);
 
                 AddCompilationRootsForRuntimeExports(module);
-           }
+            }
+
+            if (_options.MultiFile && StartupCodeMain == null)
+            {
+                foreach (var inputFile in _typeSystemContext.InputFilePaths)
+                {
+                    var module = _typeSystemContext.GetModuleFromPath(inputFile.Value);
+
+                    foreach (TypeDesc type in module.GetAllTypes())
+                    {
+                        if (!(type is EcmaType))
+                            continue;
```

Comment:
Under what conditions will GetAllTypes() return something that is not an EcmaType?